sirepo/srunit.py

```diff
@@ -478,20 +470,28 @@ def sr_sbatch_animation_run(self, sim_name, compute_model, reports, **kwargs):
             **kwargs,
         )
 
-    def sr_sbatch_login(self, compute_model, data):
+    def sr_sbatch_login(self, compute_model, sim_name):
+        from pykern.pkunit import pkexcept
         import getpass
 
+        if self.sr_sbatch_logged_in:
+            return
+        d = self.sr_sim_data(sim_name)
+        with pkexcept("SRException.*no-creds"):
+            # Must try to run sim first to seed job_supervisor.db
+            self.sr_run_sim(d, compute_model, expect_completed=False)
         p = getpass.getuser()
         self.sr_post(
             "sbatchLogin",
             PKDict(
                 password=p,
                 report=compute_model,
-                simulationId=data.models.simulation.simulationId,
-                simulationType=data.simulationType,
+                simulationId=d.models.simulation.simulationId,
+                simulationType=d.simulationType,
                 username=p,
             ),
         )
+        self.sr_sbatch_logged_in = True
```

Comment:
The post could possibly fail with data.error? Maybe assert_success() would be good to call.

Reply:
5ce403daf